tests/test_api_evolution_traversal.py

```
@@ -0,0 +1,57 @@
+        except ValueError as e:
+            assert "run_id" in str(e) or "traversal sequence" in str(e)
+        except HTTPException:
+            # This is what happens when it's vulnerable (it tries to access the path and fails with 404)
+            pytest.fail("Vulnerable: Traversal allowed in get_run_evolution_patches (got 404 instead of ValueError)")
+
+        # Test get_evolution_patch_details
+        try:
+            await evolution.get_evolution_patch_details(run_id="../etc", patch_id="patch")
+            pytest.fail("Should have raised ValueError for traversal in get_evolution_patch_details")
+        except ValueError as e:
+            assert "run_id" in str(e) or "traversal sequence" in str(e)
```

Ruff (0.14.14)

[warning] Found assertion on exception `e` in `except` block, use `pytest.raises()` instead

(PT017)